libraries/rush-lib/src/api/SubspaceConfiguration.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { FileSystem, JsonFile, JsonSchema } from '@rushstack/node-core-library';
+import schemaJson from '../schemas/subspaces.schema.json';
+import path from 'path';
+import { trueCasePathSync } from 'true-case-path';
+import { RushConfiguration } from './RushConfiguration';
+
+export interface ISubspaceConfig {
+  subspaceName: string;
+}
+
+/**
+ * This represents the JSON data structure for the "subspaces.json" configuration file.
+ * See subspace.schema.json for documentation.
+ */
+export interface ISubspaceConfigurationJson {
+  enabled: boolean;
+  splitWorkspaceCompatibility?: boolean;
+  availableSubspaces: ISubspaceConfig[];
+}
+
+/**
+ * This represents the subspace configurations for a repository, based on the "subspaces.json"
+ * configuration file.
+ * @beta
+ */
+export class SubspaceConfiguration {
+  private static _jsonSchema: JsonSchema = JsonSchema.fromLoadedObject(schemaJson);
+  /**
+   * The absolute path to the "subspaces.json" configuration file that was loaded to construct this object.
+   */
+  public readonly subspaceJsonFile: string;
+
+  /**
+   * Gets the JSON data structure for the "subspaces.json" configuration file.
+   *
+   * @internal
+   */
+  public readonly configuration: Readonly<ISubspaceConfigurationJson>;
+
+  /**
+   * A set of the available subspaces
+   */
+  public readonly availableSubspaceSet: Set<string>;
+
+  private constructor(subspaceJsonFilename: string) {
+    this.configuration = {
+      enabled: false,
+      availableSubspaces: []
+    };
+    this.subspaceJsonFile = subspaceJsonFilename;
+    this.availableSubspaceSet = new Set();
+
+    if (FileSystem.exists(this.subspaceJsonFile)) {
+      this.configuration = JsonFile.loadAndValidate(this.subspaceJsonFile, SubspaceConfiguration._jsonSchema);
+    }
+
+    for (const { subspaceName } of Object.values(this.configuration.availableSubspaces)) {
+      this.availableSubspaceSet.add(subspaceName);
```

Comment:
Report an error if the subspaceName doesn't conform to the usual `/^[a-z][a-z0-9]*([-][a-z0-9]+)*$/` pattern.

Reply:
added